### PR TITLE
Flutter package actions (#515).

### PR DESCRIPTION
**Pull request by pq**
* adds Tools menu actions for `flutter packages get` and `flutter packages upgrade`
* rejiggers actions a bit and adds a base class to do away with some duplication

Fixes: #515.

screen shot 2016-12-12 at 8 44 37 am

/cc @devoncarew

### CODE REVIEW

File: src/io/flutter/actions/FlutterPackagesGetAction.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+public class FlutterPackagesGetAction extends FlutterSdkAction {
+
+  @Override
+  public void perform(@NotNull FlutterSdk sdk, @NotNull Project project, AnActionEvent event) throws ExecutionException {
+    final Pair<Module, VirtualFile> pair = getModuleAndPubspecYamlFile(project, event);
+    if (pair != null) {
+      sdk.run(FlutterSdk.Command.PACKAGES_GET, pair.first, pair.second.getParent(), null);
+    }
+    else {
+      sdk.runProject(project, "Flutter packages get", "packages", "get");
```
Comment (devoncarew) [inline src/io/flutter/actions/FlutterPackagesGetAction.java]: This needs to be an error - `flutter packages get` requires a flutter project to work on (`flutter doctor` does not; for `flutter upgrade` its optional).

File: src/io/flutter/actions/FlutterPackagesUpgradeAction.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+public class FlutterPackagesUpgradeAction extends FlutterSdkAction {
+  @Override
+  public void perform(@NotNull FlutterSdk sdk, @NotNull Project project, AnActionEvent event) throws ExecutionException {
+    final Pair<Module, VirtualFile> pair = getModuleAndPubspecYamlFile(project, event);
+    if (pair != null) {
+      sdk.run(FlutterSdk.Command.PACKAGES_UPGRADE, pair.first, pair.second.getParent(), null);
+    }
+    else {
+      sdk.runProject(project, "Flutter packages upgrade", "packages", "upgrade");
```
Comment (devoncarew) [inline src/io/flutter/actions/FlutterPackagesUpgradeAction.java]: Same - we should show an error here.

**Comment by devoncarew** [comment]
lgtm once `packages get` and `packages upgrade` show errors if we can't resolve the best pubspec to use.

**Comment by pq** [comment]
Fixed now.  Thanks!

**Pull request merged**